simde/x86/avx512f.h

```diff
@@ -605,6 +625,38 @@ simde_mm512_castsi512_si256 (simde__m512i a) {
 #  define _mm512_castsi512_si256(a) simde_mm512_castsi512_si256(a)
 #endif
 
+SIMDE__FUNCTION_ATTRIBUTES
+simde__m512i
+simde_mm512_load_si512 (simde__m512i const * mem_addr) {
+  simde_assert_aligned(64, mem_addr);
+
+#if defined(SIMDE_AVX512F_NATIVE)
+  return _mm512_load_si512((__m512i const*) mem_addr);
+#else
+  simde__m512i r;
+  r = *mem_addr;
+  return r;
+#endif
+}
+#if defined(SIMDE_AVX512F_ENABLE_NATIVE_ALIASES)
+#  define _mm512_load_si512(a) simde_mm512_load_si512(a)
+#endif
+
+SIMDE__FUNCTION_ATTRIBUTES
+simde__m512i
+simde_mm512_loadu_si512 (simde__m512i const * mem_addr) {
+#if defined(SIMDE_AVX512F_NATIVE)
+  return _mm512_loadu_si512((__m512i const*) mem_addr);
+#else
+  simde__m512i r;
+  r = *mem_addr;
```

Comment:
This needs to use `simde_memcpy`.  Unlike `_mm512_load_si512`, `_mm512_loadu_si512` has no alignment requirements, and `*mem_addr` requires `_Alignof(simde__m512i)` alignment, which is usually 64-byte.  x86 and AArch64 tends to be pretty forgiving about unaligned accesses, but other architectures (such as arm32) will result in errors (usually a SIGBUS, IIRC).

Reply:
Sorry, I'll correct.
I realized for the alignment: I put a simde_assert_aligned in the non-u version.

On a similar subject:-
Should I use simde_memset for intrinsics which set all bytes to the same value (sometimes even all zero)?

In avx.h why does simde_mm256_lddqu_si256 call _mm256_loadu_si256 in native?
Particularly as, although equivalent, the Intel Intrinsics Guide says that _mm256_lddqu_si256 may sometimes perform better.
(Why the hell then did Intel put both in the same AVX version?)
Maybe the lddqu version isn't supported by some compilers.
How do you handle this in general?
I note that e.g. _mm256_set_m128i isn't supported by my gcc version but is OK with clang.
Are there many like this?

Reply:
> Should I use simde_memset for intrinsics which set all bytes to the same value (sometimes even all zero)?

IIRC I usually do that for one version (like *_ps), then implement the others with a castps_*(setzero_ps()).  But yeah, memset would work.

> Particularly as, although equivalent, the Intel Intrinsics Guide says that _mm256_lddqu_si256 may sometimes perform better.

I don't remember that being intentional; it should probably be reversed.

> (Why the hell then did Intel put both in the same AVX version?)

IIRC the syntax in assembly is a bit different, but I'd have to look.  Even if it were, I don't understand why there would be a performance difference; it seems like they should be able to compile to the same microcode…

> How do you handle this in general?

Case-by-case basis.  Luckily there aren't many.

> I note that e.g. _mm256_set_m128i isn't supported by my gcc version but is OK with clang.
> Are there many like this?

A few, yes.  That's one reason our CI is so extensive.  The real fun part is trying to figure out when a compiler added a function so we can support it natively in newer versions of the compiler.  IIRC you said you were using fairly old compilers; I suggest using [Compiler Explorer](https://www.godbolt.org) to try to figure that out, and hope you don't run into issues with compilers not on Compiler Explorer.  If you do, let me know; I have a lot of compilers installed locally and might be able to help.